cloud/cloud.js

```diff
@@ -856,17 +858,34 @@ Moralis.Cloud.define(
 
 /* ------------------------------------------------------------------- */
 
-Moralis.Cloud.define(
-  "checkRequesterCreatedTask",
-  async (request) => {
-    const ethAddress = request.user.get("ethAddress");
-    return await checkRequesterCreatedTask(ethAddress, request.params.taskId);
-  },
-  {
-    fields: ["taskId"],
-    requireUser: true,
+Moralis.Cloud.define("withdrawTaskerTask", async (request) => {
+  async function updateTaskStatus(taskerId, taskId) {
+    const tableName = "TaskUsers";
+
+    const TaskUsers = Moralis.Object.extend(tableName);
+    const query = new Moralis.Query(TaskUsers);
+
+    const res = await query
+      .equalTo("taskerId", taskerId)
+      .equalTo("taskId", taskId)
+      .first();
+
+    if (!res) return;
+
+    res.set("status", 4);
+    await res.save();
   }
-);
+
+  const taskId = request.params.taskId;
+  const ethAddress = request.user.get("ethAddress");
+
```

Comment:
Before calling `updateTaskStatus`, there should be checks:

1. Check that the tasker has actually claimed the task

```javascript
if (!(await checkTaskerClaimedTask(ethAddress, taskId)))
      return {
        success: false,
        message: `Address ${ethAddress} has not claimed task ${taskId}.`,
      };
```

2. Check that the current TaskUsers `status` value is `2` (ready to withdraw), and if not, do not proceed